bigquery/google/cloud/bigquery/job.py

```diff
@@ -2703,6 +2716,66 @@ def steps(self):
                 for step in self._properties.get('steps', [])]
 
 
+class TimelineEntry(object):
+    """TimelineEntry represents progress of a query job at a particular
+    point in time.
+
+    See
+    https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs
+    for the underlying API representation within query statistics.
+
+    """
+
+    def __init__(self):
+        self._properties = {}
+
+    @classmethod
+    def from_api_repr(cls, resource):
+        """Factory: construct instance from the JSON repr.
+
+        Args:
+            resource(Dict[str: object]):
+                QueryTimelineSample representation returned from API
+
+        Returns:
+            google.cloud.bigquery.TimelineEntry:
+                Timeline sample parsed from ``resource``
+        """
+        entry = cls()
+        entry._properties = resource
+        return entry
+
+    @property
+    def elapsed_ms(self):
+        """Union[int, None]: Milliseconds elapsed since start of query
+        execution."""
+        return self._properties.get('elapsedMs')
```

Comment:
Needs `_int_or_none()` call since it's an `int64` in the REST API. Probably ditto for the other ints.

https://github.com/GoogleCloudPlatform/google-cloud-python/blob/3582164c39d53bccb6aa794814a913d70adf6df6/bigquery/google/cloud/bigquery/_helpers.py#L474

Reply:
Revisited guard checks through previous PR as well.